app/src/main/java/com/kms/cura/view/fragment/PatientHomeFragment.java

```diff
@@ -0,0 +1,211 @@
+        //Place holder for speciality
+        speciality = new ArrayList<>();
+        speciality.add("A");
+        speciality.add("B");
+        speciality.add("C");
+        speciality.add("Please choose");
+        //------------------
+        specialityAdapter = new CheckBoxAdapter(getActivity(),R.layout.check_box_item,speciality,specialitySelected,updateSpinner);
+        spnSpeciality.setAdapter(specialityAdapter);
+        spnSpeciality.setSelection(specialityAdapter.getCount());
+    }
+
+    @Override
+    public void onCheckedChanged(RadioGroup group, int checkedId) {
+        switch (checkedId) {
+            case 1:
+                edtLocation.setEnabled(false);
+                int currentapiVersion = android.os.Build.VERSION.SDK_INT;
+                if (currentapiVersion > Build.VERSION_CODES.LOLLIPOP) {
+                    requestPermission();
+                } else {
+                    getCurrentLocation();
+                }
+                break;
+            case 2:
+                checked = false;
+                edtLocation.setEnabled(true);
+                break;
+            default:
+                break;
+        }
+    }
+
+    public String getAddress(double latitude, double longitude) {
+        Geocoder geocoder = new Geocoder(mContext, Locale.getDefault());
+        try {
+            List<Address> addresses = geocoder.getFromLocation(latitude, longitude, 1);
+            StringBuilder builder = new StringBuilder();
+            builder.append(addresses.get(0).getLocality());
+            builder.append(", ");
+            builder.append(addresses.get(0).getCountryName());
+            return builder.toString();
+        } catch (IOException e) {
+            ErrorController.showDialog(mContext, mContext.getResources().getString(R.string.AddressError));
+            return null;
+        }
+    }
+
+    public void requestPermission() {
+        if (ContextCompat.checkSelfPermission(activity, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            getCurrentLocation();
+            return;
+        }
+        if (ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.ACCESS_FINE_LOCATION)) {
+
+        } else {
+            requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, Settings.MY_PERMISSION_LOCATION);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case Settings.MY_PERMISSION_LOCATION: {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    getCurrentLocation();
+                }
+                return;
+            }
+        }
+    }
+
+    public void getCurrentLocation() {
```

Comment:
this method doing too much then just getCurrentLocation. It get the new address, set it to the currentLocation field, change edtLocation text and open dialog if error happened. Either you have to separate other logics out of this method, or change it name to reflect it behaviour correctly